modules/context-based-restrictions/src/test/java/com/ibm/cloud/platform_services/context_based_restrictions/v1/ContextBasedRestrictionsIT.java

```diff
@@ -0,0 +1,805 @@
+                fail(String.format("Service returned status code %d: %s%nError details: %s",
+                        e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+    @Test(groups = "replaceZone", dependsOnGroups = "getZone")
+    public void testReplaceZone() throws Exception {
+        try {
+            AddressIPAddress addressModel = new AddressIPAddress.Builder()
+                    .type("ipAddress")
+                    .value("169.23.56.234")
+                    .build();
+
+            ReplaceZoneOptions replaceZoneOptions = new ReplaceZoneOptions.Builder()
+                    .zoneId(zoneID)
+                    .ifMatch(zoneRev)
+                    .name("SDK TEST - an example of updated zone")
+                    .accountId(testAccountID)
+                    .description("SDK TEST - this is an example of updated zone")
+                    .addresses(new java.util.ArrayList<Address>(java.util.Arrays.asList(addressModel)))
+                    .transactionId(getTransactionID())
+                    .build();
+
+            // Invoke operation
+            Response<OutZone> response = service.replaceZone(replaceZoneOptions).execute();
+            // Validate response
+            assertNotNull(response);
+            assertEquals(response.getStatusCode(), 200);
+
+            OutZone outZoneResult = response.getResult();
+
+            assertNotNull(outZoneResult);
+        } catch (ServiceResponseException e) {
+            fail(String.format("Service returned status code %d: %s%nError details: %s",
+                    e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+
+    @Test(groups = "replaceZone", dependsOnMethods = "testReplaceZone")
+    public void testReplaceZoneNotFound() throws Exception {
+        try {
+            AddressIPAddress addressModel = new AddressIPAddress.Builder()
+                    .type("ipAddress")
+                    .value("169.23.56.234")
+                    .build();
+
+            ReplaceZoneOptions replaceZoneOptions = new ReplaceZoneOptions.Builder()
+                    .zoneId(NonExistentID)
+                    .ifMatch("abc")
+                    .name("SDK TEST - an example of zone")
+                    .accountId(testAccountID)
+                    .description("SDK TEST - this is an example of zone")
+                    .addresses(new java.util.ArrayList<Address>(java.util.Arrays.asList(addressModel)))
+                    .transactionId(getTransactionID())
+                    .build();
+
+            // Invoke operation
+            Response<OutZone> response = service.replaceZone(replaceZoneOptions).execute();
+        } catch (ServiceResponseException e) {
+            if (e.getStatusCode() != 404)
+                fail(String.format("Service returned status code %d: %s%nError details: %s",
+                        e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
```

Comment:
The Go and Node SDK contains tests covering `400` and `412`. I'd like to ask you to add tests here covering these status codes.

Reply:
Same as above for the 400, but the 412 I have added